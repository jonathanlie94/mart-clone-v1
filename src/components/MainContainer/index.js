import React from 'react';
import PropTypes from 'prop-types';
import Wrapper from './Wrapper';

const MainContainer = props => (
  <Wrapper>
    {props.children}
  </Wrapper>
);

MainContainer.propTypes = {
  children: PropTypes.node,
};

MainContainer.defaultProps = {
  children: false,
};

export default MainContainer;
