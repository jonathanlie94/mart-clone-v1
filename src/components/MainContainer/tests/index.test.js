import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import 'jest-styled-components';
import MainContainer from '../';

test('should render children when passed in', () => {
  const renderedComponent = shallow(
    <MainContainer>
      <div className="unique-class-name" />
    </MainContainer>
  );
  expect(
    renderedComponent.contains(<div className="unique-class-name" />)
  ).toEqual(true);
});

test('should match previous snapshot', () => {
  const listItem = shallow(<MainContainer />).dive();
  const tree = toJSON(listItem);
  expect(tree).toMatchStyledComponentsSnapshot();
});
