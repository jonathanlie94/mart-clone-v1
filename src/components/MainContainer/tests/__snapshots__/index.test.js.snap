// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match previous snapshot 1`] = `
.bLBEqh {
  padding: 0.75rem;
}

<div
  className="sc-bdVaJa bLBEqh"
/>
`;
